lib/core/src/recover/recoverer.rs

```diff
@@ -280,9 +313,23 @@ impl Recoverer {
                 Swap::Chain(chain_swap) => match chain_swap.direction {
                     Direction::Incoming => {
                         let Some(recovered_data) = recovered_chain_receive_data.get(swap_id) else {
-                            log::warn!("Could not apply recovered data for incoming Chain swap {swap_id}: recovery data not found");
+                            warn!("Could not apply recovered data for incoming Chain swap {swap_id}: recovery data not found");
                             continue;
                         };
+
+                        let claim_is_cleared = chain_swap.claim_tx_id.is_some()
+                            && recovered_data.lbtc_claim_tx_id.is_none();
+                        let refund_is_cleared = chain_swap.refund_tx_id.is_some()
+                            && recovered_data.btc_refund_tx_id.is_none();
+                        if is_local_within_grace_period && (claim_is_cleared || refund_is_cleared) {
```

Comment:
I still think it might miss the case where we just broadcast a second refund but only the first is seen by the recoverer (It will revert to the first refund_tx_id and state Failed briefly).

Reply:
@dangeross wouldn't this be a temporary situation? Wouldn't it be fixed as soon as the previous (overridden)  refund will get out of the mempool?

Reply:
Yeah, it's temporary. It might just be for one sync cycle

Reply:
> I still think it might miss the case where we just broadcast a second refund but only the first is seen by the recoverer (It will revert to the first refund_tx_id and state Failed briefly).

Yeah, I agree that might happen. Maybe it's fine for now. I don't see a simple way of addressing that issue. Maybe the solution is having a more complete and complex model that tracks multiple refunds separately